### PR TITLE
Added support for I2C power control during boot2 phase

### PR DESCRIPTION
**Pull request by skieast**
Added support for I2C TCA9554 peripheral power control during the boot2 phase.
Uses software I2C. 
Made a small change to the neopixel reset timing (increased to 1000usec) as it seems to be needed on this board.
I tested this timing with another neopixel equipped board (Gravitech Cucumber) and it worked.

### CODE REVIEW

**Comment by skieast** [comment]
image

Initializing the TCA9554 using software I2C

File: ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c
Comment (hathach) [inline ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c:526]: is there any reason that you move the delay before the neopixel off ? would the pixel is off when the delay is as it is ?

**Review by hathach — COMMENTED** [review]
I have made some changes to the PR, but couldn't push it to your fork to update this PR due to permission. Please set your fork to allow maintainer to have write permission, or simply grant me write permission

File: ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c
Comment (skieast) [reply ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c:526]: I'll wait until you have pushed whatever changes you have made.  You are correct in that it works either way and should be after.  This is a remnant from trying to eliminate what I thought was a problem. Thanks

**Comment by skieast** [comment]
> 
> 
> I have made some changes to the PR, but couldn't push it to your fork to update this PR due to permission. Please set your fork to allow maintainer to have write permission, or simply grant me write permission

I have invited you as a collaborator on my fork. Hopefully this is what you are asking for. Thanks for looking at the PR

**Comment by hathach** [comment]
> > I have made some changes to the PR, but couldn't push it to your fork to update this PR due to permission. Please set your fork to allow maintainer to have write permission, or simply grant me write permission
> 
> I have invited you as a collaborator on my fork. Hopefully this is what you are asking for. Thanks for looking at the PR

thanks for the invitation, actually, I just fiugred out I made an typo in the push command previously. Anyway, I pushed the changes, mostly some format clean up along with the macro for NEOPIXEL_RESET_DELAY set to 1000 us for using gpio expander, 200us for existing boards.

File: ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c
Comment (hathach) [inline ports/esp32s2/components/bootloader/subproject/main/bootloader_start.c]: the delay should be below its comment. But I push update for this change already.

**Review by hathach — APPROVED** [review]
Thank you for your PR, everything look perfect now. Will merge when ci is passed.

**Pull request merged**
